### PR TITLE
City construction queue

### PR DESCRIPTION
**Pull request by r3versi**
feature

I changed both the backend to support construction queues and the `CityScreen` UI to gain some space to show the queue.

* Added `constructionQueue` in addition to `currentConstruction`: if `currentConstruction` is done, next construction from Queue is started; if Queue is empty invoke `ConstructionAutomation`
* Queue utility methods: add, remove, higher prio, lower prio
* Icons to move constructions in queue
* Top left city stats moved to top right panel
* Added current construction and queue construction to top left
* Extended selected construction (containing description) moved to bottom right, it is now displayed alternatively to selected tile
* Max queue size = 10

**Looking for testers!**

### CODE REVIEW

File: core/src/com/unciv/logic/city/CityConstructions.kt [hunk 1]
```diff
@@ -252,16 +283,66 @@ class CityConstructions {
...
+            if (currentConstruction != "") return
+        }
+
         ConstructionAutomation(this).chooseNextConstruction()
     }
-    //endregion
 
+    fun isEnqueuable(constructionName: String): Boolean {
+        return true
+    }
```
Comment (naivekook) [inline core/src/com/unciv/logic/city/CityConstructions.kt]: is this ok? looks like useless function

**Comment by r3versi** [comment]
Oops. I'll clean this. `isEnqueuable` is not needed in the last version
implemented.

Il giorno sab 28 dic 2019 alle ore 13:49 Vladimir Tanakov <
notifications@github.com> ha scritto:

File: core/src/com/unciv/logic/city/CityConstructions.kt (hunk 1 above)
Comment (r3versi) [reply core/src/com/unciv/logic/city/CityConstructions.kt]: Oops, I'll clean this. It isn't needed in last implementation

File: core/src/com/unciv/ui/cityscreen/CityInfoTable.kt [hunk 2]
```diff
@@ -92,6 +96,65 @@ class CityInfoTable(private val cityScreen: CityScreen) : Table(CameraStageBaseS
...
+                        / cityInfo.cityStats.currentCityStats.food).toInt()
+                if (turnsToPopulation < 1) turnsToPopulation = 1
+                turnsToPopString = "[$turnsToPopulation] turns to new population".tr()
+            }
+        } else if (cityInfo.cityStats.currentCityStats.food < 0) {
+            val turnsToStarvation = floor(cityInfo.population.foodStored / -cityInfo.cityStats.currentCityStats.food).toInt() + 1
+            turnsToPopString = "[$turnsToStarvation] turns to lose population".tr()
+        } else {
+            turnsToPopString = "Stopped population growth".tr()
+        }
```
Comment (yairm210) [inline core/src/com/unciv/ui/cityscreen/CityInfoTable.kt]: Should probably be "when"

File: core/src/com/unciv/ui/cityscreen/CityInfoTable.kt [hunk 3]
```diff
@@ -92,6 +96,65 @@ class CityInfoTable(private val cityScreen: CityScreen) : Table(CameraStageBaseS
...
+        statsTable.add(Label(turnsToPopString + " (" + cityInfo.population.foodStored + "/" + cityInfo.population.getFoodToNextPopulation() + ")"
```
Comment (yairm210) [inline core/src/com/unciv/ui/cityscreen/CityInfoTable.kt]: val foodLabelText = ...
statsTable.add(foodLabelText)

**Comment by yairm210** [comment]
Sorry, now I noticed that what I commented on was actually from a previous place so it's not against your change
This looks really good!
You appear to have some conflicts, if you resolve them then we'll add this in!

**Pull request merged**